### ISSUE TITLE
YubiKey support

### ISSUE THREAD

**Issue by codebam** [termux/termux-api#393]
**Feature description**
Support for YubiKey USB devices in OpenPGP.

**Reference implementation**
<https://github.com/open-keychain/open-keychain>

Does another app/terminal emulator have this feature?
OpenKeyChain has this feature on Android

Provide links to more background information

Although OpenKeyChain supports this feature, the PGP implementation in OpenKeyChain doesn't support YubiKeys in such a way that it can decrypt files with a hidden receipt. This creates a problem with password-store (and the password store app) because password store encrypts keys such that the recipient is hidden. See [here](https://github.com/zeapo/Android-Password-Store/issues/173).

OpenPGP doesn't have this issue, and supporting it would open up possibilities to do signing and key verification on git and any other applications that support PGP.

**Comment by Fresheyeball** [comment]
Please?

**Comment by sn0n** [comment]
Following for updates

**Comment by Grimler91** [comment]
For the record things like gpg signing and decrypting with a yubikey works fine when running as root. I haven't gotten signing of git commits working though, might need some patch to modify how git invokes gpg.

Adding full support for all yubikey features, for all software might be hard, but we should be able to make softwares that make use of libusb able to (partially) access yubikeys, without root

**Comment by inducer** [comment]
@calaveraInfo Thank you so much for that tip! I have my SSH keys on Yubikeys following https://github.com/drduh/YubiKey-Guide/blob/master/README.md, and OpenKeychain plus okc-agent worked for me on an AOSP-only phone (LineageOS, but no GMS).

**Comment by sevmonster** [comment]
OpenKeychain is [no longer supported](https://github.com/open-keychain/open-keychain) and OkcAgent [no longer works](https://github.com/DDoSolitary/OkcAgent/issues/143). A native solution to this would be much appreciated.

**Comment by juliuskreutz** [comment]
Following for updates

**Comment by gordonswing** [comment]
Desperately waiting for the native solution as well

**Comment by ludwig234** [comment]
I would love if termux supported FIDO2 "*-sk" SSH keys.
If that was the case I could replace all other sash keys with the FIDO one.

**Comment by KnifeOfDunwall** [comment]
Any updates on this? Ive tried for hours to get the okc-agent and okc to work but now im stuck with a bug where okc doesnt transmit the decoded keys correctly, making the decoded one useless. ive seen someone wanting to make a ssh agent for this but the github repo has one commit with all functions declared but in it theres just basically "print('not implemented')"

**Comment by sevmonster** [comment]
No maintainers are interested, and no one else has stepped forward yet. Be the change you want 🙂

Both OpenKeychain and OkcAgent are no longer being updated, so ultimately both of them need to be replaced. The task is substantial.

**Comment by sevmonster** [comment]
Are you trying to make me look like a liar...?

**Comment by agnostic-apollo** [comment]
And whose fault is that, you were the one making assumptions that no maintainer was interested ;)

**Comment by sevmonster** [comment]
Well I've never been more pleased to be proven wrong :)

**Comment by uplight-dev** [comment]
any updates on this? it's been a month already

**Comment by 1ynxy** [comment]
Absolutely no pressure, but is this still likely to happen within a couple of months or so? It would be a massive improvement over the existing work-arounds, which I'm struggling to get working.

**Comment by agnostic-apollo** [comment]
Yes

**Comment by 1ynxy** [comment]
Sorry to be a pest, but does the latest referenced commit get us any closer to Yubikey support inside Termux? I'm unsure as to whether it has even been upstreamed yet, but it does reference this issue and claim to implement a fix.

**Comment by robertkirkman** [comment]
No, that's someone's personal fork. They are merging a lot of PRs in it to experiment with. Let's call this version  "billybox1926-jpg `com.termux.api.debug`" because it names itself that a lot in its code at first glance.

https://github.com/billybox1926-jpg/termux-api

**Comment by robertkirkman** [comment]
It looks like it is clearly labeled with explanations about what it is and how to use it,

<img width="1406" height="164" alt="Image" src="https://github.com/user-attachments/assets/28d91134-c119-418b-a3e2-3b67c5914acf" />

They do have a Termux fork as well with a lot of changes in it. I would say that someone should probably not use it unless they are acquainted with the billybox1926-jpg termux ecosystem and have personally audited it to check for unwanted code.

**Comment by sevmonster** [comment]
#894 for reference
